content/ui-testing-handbook/react/en/interaction-testing.md

```diff
@@ -0,0 +1,241 @@
+
+  it('should pin a task', async () => {
+    const { queryByText, getByRole } = render(<Default />);
+
+    await waitFor(() => {
+      expect(queryByText('You have no tasks')).not.toBeInTheDocument();
+    });
+
+    const getTask = () => getByRole('listitem', { name: 'Export logo' });
+
+    const pinButton = within(getTask()).getByRole('button', { name: 'pin' });
+
+    fireEvent.click(pinButton);
+
+    const unpinButton = within(getTask()).getByRole('button', {
+      name: 'unpin',
+    });
+
+    expect(unpinButton).toBeInTheDocument();
+  });
+});
+```
+
+Run `yarn test` to start up Jest. You’ll notice that the test fails.
+
+![](/ui-testing-handbook/yarn-test-fail.png)
+
+`InboxScreen` fetches data from the back-end. In the [previous chapter](../composition-testing), we set up [Storybook MSW addon](https://storybook.js.org/addons/msw-storybook-addon/) to mock this API request. However, that isn’t available to Jest. We’ll need a way to bring this and other component dependencies along.
+
+### Component configs to go
+
+Complex components rely on external dependencies such as theme providers and context to share global data. Storybook uses [decorators](https://storybook.js.org/docs/react/writing-stories/decorators) to wrap a story and provide such functionality. To import stories along with all their config, we'll use the [@storybook/testing-react](https://github.com/storybookjs/testing-react) library.
+
+This is usually a two-step process. First, we need to register all global decorators. In our case, we have two: a decorator that provides the Chakra UI theme and one for the MSW addon. Both configured via the [`.storybook/preview`](https://github.com/chromaui/ui-testing-guide-code/blob/interaction-testing/.storybook/preview.js) file.
+
+Jest offers a global setup file `setupTests.js`, auto-generated by CRA when the project is bootstrapped. Update that file to register Storybook’s global config.
+
+```javascript:title=setupTests.js
+import '@testing-library/jest-dom';
+
+import { setGlobalConfig } from '@storybook/testing-react';
+import * as globalStorybookConfig from '../.storybook/preview';
+
+setGlobalConfig(globalStorybookConfig);
+```
+
+Next, update the test to use the `composeStories` utility from `@storybook/testing-react`. It returns a 1:1 map of the stories with all decorators applied to them. And voilà, our test is passing!
+
+```javascript:title=src/InboxScreen.test.js
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, waitFor, cleanup } from '@testing-library/react';
+import { composeStories } from '@storybook/testing-react';
+import { getWorker } from 'msw-storybook-addon';
+import * as stories from './InboxScreen.stories';
+
+describe('InboxScreen', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  // Clean up after all tests are done, preventing this
+  // interception layer from affecting irrelevant tests
+  afterAll(() => getWorker().close());
```

Comment:
If you're going to introduce `setupTests.js`, you could do this (as well as import `@testing-library/jest-dom/extend-expect`) in that file, instead.